python/pyarrow/tests/test_dataset.py

```diff
@@ -623,7 +626,7 @@ def test_partitioning():
         dictionaries={
             "key": pa.array(["first", "second", "third"]),
         })
-    assert partitioning.dictionaries[0].to_pylist() == [
+    assert partitioning.dictionaries[1].to_pylist() == [
```

Comment:
Can you also assert here that the first element is `None` (`assert partitioning.dictionaries[0] is None`)

Reply:
Made the change.